chromadb/db/clickhouse.py

```diff
@@ -445,6 +445,12 @@ def get_nearest_neighbors(
         collection_name=None,
         collection_uuid=None,
     ) -> Tuple[List[List[uuid.UUID]], List[List[float]]]:
+        
+        if collection_name is not None:
+            collection_uuid = self.get_collection_uuid_from_name(collection_name)
+
+        self._idx.load_if_not_loaded(collection_uuid)
```

Comment:
We run into trouble because we only load this on `get_nearest_neighbors`. I think we should instead do this on a load of a collection. `get_nearest_neighbors` should error if we got this far without having an index loaded.

Reply:
Probably OK until we do the refactor for swapping indices.

Reply:
Yeah, I think the real answer is we need to handling the bookkeeping of indexes better.